kirin/piv/model_maker.py

```diff
@@ -388,29 +390,25 @@ def _make_trip_update(self, json_train, vj):
                 if event is None:
                     continue
 
-                piv_disruption = get_value(event, "evenement", nullable=True)
-
-                piv_event_status = get_value(event, "statutModification", nullable=True)
-                if not piv_event_status and piv_disruption:
-                    piv_event_status = get_value(piv_disruption, "type")
+                piv_disruption = event.get("evenement", {})
+                piv_event_status = event.get("statutModification", piv_disruption.get("type", None))
 
                 if not piv_event_status or piv_event_status in MANAGED_STOP_EVENTS:
                     piv_event_datetime = get_value(event, "dateHeureReelle", nullable=True)
                     event_datetime = as_utc_naive_dt(piv_event_datetime) if piv_event_datetime else None
                     if event_datetime:
                         rt_stop_time[event_toggle] = event_datetime
                         setattr(st_update, STOP_EVENT_DATETIME_MAP[event_toggle], event_datetime)
-                    if piv_disruption:
+                    if piv_disruption and piv_event_status in ["RETARD_OBSERVE", "RETARD_PROJETE"]:
                         piv_event_delay = piv_disruption.get("retard", {}).get("duree", 0)
                         setattr(st_update, DELAY_MAP[event_toggle], as_duration(piv_event_delay * 60))  # minutes
```

Comment:
minor: I would separate both (and too bad if it's not consistent in the feed)
```suggestion
                    if piv_disruption:
                        piv_event_delay = piv_disruption.get("retard", {}).get("duree", 0)

                    if piv_event_status in ["RETARD_OBSERVE", "RETARD_PROJETE"]:
                        setattr(st_update, DELAY_MAP[event_toggle], as_duration(piv_event_delay * 60))  # minutes
```

Reply:
if not piv_disruption and  piv_event_status in ["RETARD_OBSERVE", "RETARD_PROJETE"], `piv_event_delay` will be not referenced

Reply:
done in c0ebe4d0a59968277aaf7de7e7a45fe34911248b